.github/workflows/autopilot-branch-to-latest.yaml

```diff
@@ -0,0 +1,254 @@
+  # perform updates to all nodes.
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs: [ setup ]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # The 'testdata' matrix focuses on grouping updates of individual historic branch releases
+        # to the latest released version on a branch.
+        testdata: [
+          # v1.24 versions
+          {
+            name: "v1242-0",
+            fromver: "v1.24.2+k0s.0",
+            tover: "${{ github.event.inputs.v124latest }}",
+            subnet: 0,
+          },
+          {
+            name: "v1243-0",
+            fromver: "v1.24.3+k0s.0",
+            tover: "${{ github.event.inputs.v124latest }}",
+            subnet: 1,
+          },
+          {
+            name: "v1244-0",
+            fromver: "v1.24.4+k0s.0",
+            tover: "${{ github.event.inputs.v124latest }}",
+            subnet: 2,
+          },
+
+          # latest branch versions to latest
+          {
+            name: "v124to125",
+            fromver: "${{ github.event.inputs.v124latest }}",
+            tover: "${{ github.event.inputs.v125latest }}",
+            subnet: 20,
+          },
+        ]
+
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v2
+      with:
+        terraform_wrapper: false
+        terraform_version: ${{ env.TERRAFORM_VERSION }}
+
+    - name: Cache Tool Image (restore)
+      uses: actions/cache@v3
+      with:
+        key: tool-${{ github.sha }}
+        path: tool-latest.tar.gz
+
+    - name: Import Tool
+      run: |
+        docker load --input tool-latest.tar.gz
+        rm tool-latest.tar.gz
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+
+    - name: Run tool
+      run: |
+        mkdir data
+        go test -v -timeout 60m -run TestVersionUpdateSuite ./inttest/toolsuite/tests \
+          --testname k0sts-${{ matrix.testdata.name }} \
+          --operation-timeout 30m \
+          --data-dir $(pwd)/data \
+          --provider aws \
+          --command havpc -- \
+            --cluster-name k0sts-${{ matrix.testdata.name }} \
+            --controllers ${{ github.event.inputs.controllers }} \
+            --k0s-version ${{ matrix.testdata.fromver }} \
+            --region ${{ github.event.inputs.region }} \
+            --workers ${{ github.event.inputs.workers }} \
+            --k0s-update-version ${{ matrix.testdata.tover }} \
+            --vpc-id ${{ needs.setup.outputs.vpcid }} \
+            --subnet-idx ${{ matrix.testdata.subnet }}
+
+  # teardown is always run as a cleanup for the VPC infrastructure
+  teardown:
+    name: Teardown Infrastructure
+    runs-on: ubuntu-latest
+    needs: [ test ]
+    if: ${{ always() }}
```

Comment:
What about running this only if the setup step was successful?

Reply:
Wouldn't that make `teardown` run parallel with `test`?